packages/contract/src/DaimoFastCCTP.sol

```diff
@@ -0,0 +1,241 @@
+
+// Wraps CCTP. Allows optimistic fast transfers. Alice initiates a transfer by
+// calling `startTransfer` on chain A. After the CCTP delay (currently 10+ min),
+// funds arrive at the DaimoFastCCTP contract deployed on chain B. Bob can call
+// `claimTransfer` to claim the funds. Alternately, immdiately after the first
+// call, an LP can call `fastFinishTransfer` to send Bob his funds immediately.
+// Later, when the funds arrive from CCTP, the LP (rather than Bob) will be able
+// to claim.
+contract DaimoFastCCTP {
+  using SafeERC20 for IERC20;
+
+  // Commit to transfer details in a handoff address. See EphemeralHandoff.
+  mapping(address => bool) public handoffSent;
+  // On receiving chain, map each transfer to recipient (LP or Bob).
+  mapping(address => address) public handoffToRecipient;
+
+  constructor() {}
+
+  // Called by Alice on Chain A. Requires approval for (fromToken, fromAmount).
+  // Initiates a CCTP transfer.
+  function startTransfer(
+    ICCTPTokenMessenger cctpMessenger,
+    IERC20 fromToken,
+    uint256 fromAmount,
+    uint256 toChainID,
+    address toAddr,
+    uint256 toAmount,
+    IERC20 toToken,
+    uint256 nonce
+  ) public {
+    require(fromAmount >= toAmount, "FCCTP: fromAmount < toAmount");
+
+    // Deploy CCTP sender
+    uint256 fromChainID = block.chainid;
+    address fromAddr = msg.sender;
+    EphemeralHandoff handoff = new EphemeralHandoff{salt: 0}(
+      fromChainID,
+      fromAddr,
+      fromAmount,
+      toChainID,
+      toAddr,
+      toAmount,
+      toToken,
+      nonce
+    );
+
+    // Ensure we don't reuse nonces. This covers the case where Alice sends Bob
+    // the same amount twice in a row = otherwise identical transfers = no way
+    // for an LP to fastFinish() the second one without a distinct handoff addr.
+    require(
+      !handoffSent[address(handoff)],
+      "FCCTP: identical transfer already sent"
+    );
+    handoffSent[address(handoff)] = true;
+
+    // Receive funds from Alice
+    fromToken.safeTransferFrom(fromAddr, address(handoff), fromAmount);
+
+    // Initiate transfer + destroy handoff, minimizing gas consumption.
+    handoff.startTransferAndSelfDestruct(cctpMessenger, fromToken);
+  }
+
+  // Pays Bob immediately on chain B. The caller LP sends (toToken, toAmount).
+  // Later, when the slower CCTP transfer arrives, the LP will be able to claim
+  // (toToken, fromAmount), keeping the spread (if any) between the amounts.
+  function fastFinishTransfer(
+    uint256 fromChainID,
+    address fromAddr,
+    uint256 fromAmount,
+    uint256 toChainID,
+    address toAddr,
+    uint256 toAmount,
+    IERC20 toToken,
+    uint256 nonce
+  ) public {
+    // Calculate handoff address
+    EphemeralHandoff handoff = new EphemeralHandoff{salt: 0}(
+      fromChainID,
+      fromAddr,
+      fromAmount,
+      toChainID,
+      toAddr,
+      toAmount,
+      toToken,
+      nonce
+    );
+    handoff.destroy();
```

Comment:
just to confirm, you only need the address of the handoff here right? might be cleaner to make a util function for computing the create2 address or adding a comment